src/pages/ReportDetailsPage.tsx

```diff
@@ -198,6 +220,29 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
         />
     ) : null;
 
+    const renderAvatar = isGroupChat ? (
```

Comment:
It looks like there's a separate case that's not captured here for money request reports. I suggest memoizing this and capturing all the cases:

```diff
diff --git a/src/pages/ReportDetailsPage.tsx b/src/pages/ReportDetailsPage.tsx
index 4e6219500d..c4c95c22c7 100644
--- a/src/pages/ReportDetailsPage.tsx
+++ b/src/pages/ReportDetailsPage.tsx
@@ -220,27 +220,42 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
         />
     ) : null;
 
-    const renderAvatar = isGroupChat ? (
-        <AvatarWithImagePicker
-            source={icons[0].source}
-            isUsingDefaultAvatar={!report.avatarUrl}
-            size={CONST.AVATAR_SIZE.XLARGE}
-            avatarStyle={styles.avatarXLarge}
-            disableViewPhoto
-            onImageRemoved={() => {
-                // Calling this without a file will remove the avatar
-                Report.updateGroupChatAvatar(report.reportID ?? '');
-            }}
-            onImageSelected={(file) => Report.updateGroupChatAvatar(report.reportID ?? '', file)}
-            editIcon={Expensicons.Camera}
-            editIconStyle={styles.smallEditIconAccount}
-        />
-    ) : (
-        <RoomHeaderAvatars
-            icons={icons}
-            reportID={report?.reportID}
-        />
-    );
+    const avatar = useMemo(() => {
+        if (isMoneyRequestReport) {
+            return (
+                <MultipleAvatars
+                    icons={icons}
+                    size={CONST.AVATAR_SIZE.LARGE}
+                />
+            );
+        }
+
+        if (isGroupChat) {
+            return (
+                <AvatarWithImagePicker
+                    source={icons[0].source}
+                    isUsingDefaultAvatar={!report.avatarUrl}
+                    size={CONST.AVATAR_SIZE.XLARGE}
+                    avatarStyle={styles.avatarXLarge}
+                    disableViewPhoto
+                    onImageRemoved={() => {
+                        // Calling this without a file will remove the avatar
+                        Report.updateGroupChatAvatar(report.reportID ?? '');
+                    }}
+                    onImageSelected={(file) => Report.updateGroupChatAvatar(report.reportID ?? '', file)}
+                    editIcon={Expensicons.Camera}
+                    editIconStyle={styles.smallEditIconAccount}
+                />
+            );
+        }
+
+        return (
+            <RoomHeaderAvatars
+                icons={icons}
+                reportID={report?.reportID}
+            />
+        );
+    }, [icons, isGroupChat, isMoneyRequestReport, report.avatarUrl, report.reportID, styles]);
 
     const reportName = isGroupChat ? ReportUtils.getGroupChatName(undefined, true, report.reportID ?? '') : ReportUtils.getReportName(report);
     return (
@@ -253,16 +268,7 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
                 />
                 <ScrollView style={[styles.flex1]}>
                     <View style={styles.reportDetailsTitleContainer}>
-                        <View style={styles.mb3}>
-                            {isMoneyRequestReport ? (
-                                <MultipleAvatars
-                                    icons={icons}
-                                    size={CONST.AVATAR_SIZE.LARGE}
-                                />
-                            ) : (
-                                renderAvatar
-                            )}
-                        </View>
+                        <View style={styles.mb3}>{avatar}</View>
                         <View style={[styles.reportDetailsRoomInfo, styles.mw100]}>
                             <View style={[styles.alignSelfCenter, styles.w100, styles.mt1]}>
                                 <DisplayNames

```

Reply:
I like it. Let's do it after.